script.js

```diff
@@ -389,7 +389,8 @@ window.GaiaAppIcon = (function(exports) {
   };
 
   proto._relayout = function() {
-    this._size = this.clientWidth;
+    var bookmarkIcon = this.bookmark && this.bookmark.icon;
+    this._size = this.clientWidth || (bookmarkIcon && bookmarkIcon.size);
```

Comment:
I'm not certain this change makes sense - the component is designed to be added to the DOM before it gets refreshed so it can know its size (alternatively, a size can be set on it via style, I suppose?) - this just means we may end up with badly scaled icons when it does get added to the DOM (even if bookmarkIcon.size ends up larger than clientWidth, down-scaling can also look bad and wastes memory).